From: Jens Axboe <axboe@fb.com>
Date: Tue, 1 Nov 2016 10:01:35 -0600
Subject: writeback: mark background writeback as such
References: bsc#1020989,FATE#322379
Git-commit: 13edd5e7315a26b448c5f7f33fc7721b1e0c17ef
Patch-Mainline: v4.10-rc1

If we're doing background type writes, then use the appropriate
background write flags for that.

Signed-off-by: Jens Axboe <axboe@fb.com>
Reviewed-by: Christoph Hellwig <hch@lst.de>
Acked-by: Hannes Reinecke <hare@suse.com>
---
 include/linux/writeback.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/writeback.h b/include/linux/writeback.h
index 560bfe1..a0771d2 100644
--- a/include/linux/writeback.h
+++ b/include/linux/writeback.h
@@ -105,6 +105,8 @@ static inline int wbc_to_write_flags(struct writeback_control *wbc)
 {
 	if (wbc->sync_mode == WB_SYNC_ALL)
 		return REQ_SYNC;
+	else if (wbc->for_kupdate || wbc->for_background)
+		return REQ_BACKGROUND;
 
 	return 0;
 }
-- 
1.8.5.6

